recon/src/main/java/org/hps/recon/utils/TrackClusterMatcherMinDistance.java

```diff
@@ -0,0 +1,753 @@
+package org.hps.recon.utils;
+
+import org.hps.recon.ecal.cluster.ClusterUtilities;
+import org.hps.recon.ecal.cluster.ClusterCorrectionUtilities;
+import org.hps.recon.tracking.TrackUtils;
+import org.hps.recon.tracking.TrackData;
+import org.hps.record.StandardCuts;
+
+import hep.aida.IAnalysisFactory;
+import hep.aida.IHistogram1D;
+import hep.aida.IHistogram2D;
+import hep.aida.IHistogramFactory;
+import hep.aida.ITree;
+import hep.aida.ref.rootwriter.RootFileStore;
+import hep.physics.vec.BasicHep3Vector;
+import java.io.IOException;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.List;
+import java.util.ArrayList;
+
+import org.lcsim.event.Track;
+import org.lcsim.event.TrackState;
+import org.lcsim.event.Cluster;
+import org.lcsim.event.RelationalTable;
+import org.lcsim.event.EventHeader;
+import org.lcsim.event.base.BaseRelationalTable;
+import org.lcsim.event.LCRelation;
+import org.lcsim.event.base.BaseCluster;
+import org.lcsim.geometry.subdetector.HPSEcal3;
+import org.lcsim.event.ReconstructedParticle;
+
+
+/**
+ * Utility used to determine if a Tracks and Ecal Cluster are matched.
+ * Class extends AbstractTrackClusterMatcher, which implements
+ * TrackClusterMatcherInter
+ *
+ * @author <a href="mailto:alspellm@ucsc.edu">Alic Spellman</a>
+ */
+
+public class TrackClusterMatcherMinDistance extends AbstractTrackClusterMatcher{
+
+    protected HashMap<Cluster, Track> clusterToTrack;
+    protected RelationalTable hitToRotated = null;
+    protected RelationalTable hitToStrips = null;
+    protected RelationalTable trackToData = new BaseRelationalTable(RelationalTable.Mode.ONE_TO_ONE, RelationalTable.Weighting.UNWEIGHTED);
+    protected String trackClusterCollectionName = "GBLTracks";
+    protected String rootFile = String.format("%s_TrackClusterMatching.root",this.trackCollectionName);
+    //Keep false. May be used in future 03.19.21
+    protected Boolean buildParamHistos = false;
+    /*
+     * Cuts used for track+cluster residual matching 
+     */
+    protected double tcut = 8.0;
```

Comment:
These should be configurable

Reply:
Can I add the dx and dy cuts in to the StandardCuts.java file?That seems like a reasonable way to pass these

Reply:
I find that class confusing. Avoid. Here is OK.